articles/you-are-not-a-software-engineer.md

```diff
@@ -0,0 +1,58 @@
+
+## What is your drive?
+
+You don't want to just play around and not produce anything of worth to someone, I hope.
+
+According to recent models of motivation ("Drive" by Daniel Pink), our actions are fueled from autonomy, mastery, and purpose.
+
+In healthy working environments, autonomy is easy. You have the power to decide how will you implement the solution, to choose working mode which enables you to do so, and influence decisions, within your ability.
+
+Mastery is the urge to improve your skill set and be proud after all the work you have done. This moves us. To change projects, gather new experiences and work with new technologies. Since IT field is a fast-moving freight train, I assume you are already doing this.
+
+Purpose can be tricky, though. The desire to do something that is of meaning and importance, greater than ourselves. Maybe helping already successful enterprise earn 5% net profit greater than previous year is not your thing. Maybe you want to help humanity find the cure for cancer. Boarding on some ideas is easy, while others not. I will not provide an answer to this. It is entirely up to you. I will give you a hint though. It is everywhere. It involves feelings.
+
+## "I am in the business of creating value."
+
+Remember the cost of fixing a bug, depending on the stage of software development lifecycle. An hour spent in requirements engineering will save you many more down the road. You might use a framework designed to prevent this from happening, prescribing you the points in time when you should do requirements engineering. But, all the magical frameworks in the world will not help you, if you don't understand what you are building.
+
+## "I am in the business of satisfying customer."
+
+If you consider yourself as a one-man business, your customer is every person you interact with. Your customers are your team members, management, stakeholders, users. Keeping them happy is of importance for your business, even if you need to make a couple of trade-offs. Sometimes, you might feel disconnected from the customer. Either he is really busy and doesn't come by, or you see him perhaps once on a release review. Establishing that link creates a fertile ground for collaboration. Having it is your right, and responsibility.
+
+## Anybody can code
+
+After all, we live in a world where anybody can code.
+
+> "Have you heard? The other supplier was X times cheaper".
+
+All over the world, you can find coders capable of writing code. And doing it cheap!
+
+Why would the customer choose you, over any other provider?
+
+> "We use sound design principles, and our architecture is flawless!"
+
+Yes, kudos to that, you should really keep it. Selling your work like that totaly make sense - if all your customes are engineers. To others, that doesn't mean much. When the end result is the same - working software, being proud of ourselves, thinking that we can do it better than others, is not a good selling point.
```

Comment:
2 typos "Selling your work like that totaly make sense" -> "Selling your work like that _totally_ _makes_ sense"

sprachlich etwas holprig: "When the end result is the same - working software, being proud of ourselves, thinking that we can do it better than others, is not a good selling point. "
-> ist es klarer wenn umgekehrt?
"The fact that we are proud of our work is not a good selling point - considering the end result is the same: working software"

Reply:
2 typos ...

a bit bumpy formulated ... maybe change the order

Reply:
I see what you're pointing at. I adopted your proposal.